pcsx2/gui/MainFrame.cpp

```diff
@@ -643,32 +644,55 @@ void MainEmuFrame::ApplyCoreStatus()
 		}
 	}
 
-	if( cdvd )
+	wxString Source_Label[static_cast<int>(CDVD_SourceType::NumberofSources)] = {"ISO" , "CDVD" , "BIOS"};
+	wxString Reboot_Boot_Indicator = (vmIsActive) ? "Reboot" : "Boot";
+	wxString ItemLabel = Reboot_Boot_Indicator + " " + Source_Label[static_cast<int>(source)];
+	wxString Fast_Reboot_HelpText = "Reboot using fast BOOT (skips splash screens)";
+	wxString Fast_Boot_HelpText = "Use fast boot to skip PS2 startup and splash screens";
+	wxString Full_Boot_HelpText[static_cast<int>(CDVD_SourceType::NumberofSources)];
+	wxString Full_Reboot_HelpText = "Hard reset of the active VM";
+
+	// Initialize Full Boot Help text for all the source mediums
+	Full_Boot_HelpText[static_cast<int>(CDVD_SourceType::CDVDsrc_Iso)]    = "Boot the VM using the current ISO source media";
+	Full_Boot_HelpText[static_cast<int>(CDVD_SourceType::CDVDsrc_Plugin)] = "Boot the VM using the current CD/DVD source media";
+	Full_Boot_HelpText[static_cast<int>(CDVD_SourceType::CDVDsrc_NoDisc)] = "Boot the VM without any source media";
+	
+	if (cdvd)
 	{
-		if( vm )
-		{
-			cdvd->SetItemLabel(_("Reboot CDVD (f&ull)"));
-			cdvd->SetHelp(_("Hard reset of the active VM."));
-		}
+		if (source == CDVD_SourceType::CDVDsrc_NoDisc)
+			cdvd->SetItemLabel(ItemLabel);
 		else
-		{
-			cdvd->SetItemLabel(_("Boot CDVD (f&ull)"));
-			cdvd->SetHelp(_("Boot the VM using the current DVD or Iso source media"));
-		}	
+			cdvd->SetItemLabel(ItemLabel + " (f&ull)");
+
+		if (vmIsActive)
+			cdvd->SetHelp(Full_Reboot_HelpText);
+		else
+			cdvd->SetHelp(Full_Boot_HelpText[static_cast<int>(source)]);
 	}
 
-	if( cdvd2 )
+	if (cdvd2 || Destroyed_CDVD2)
```

Comment:
> It's equivalent to if (cdvd2 || !cdvd2). *sigh*

No, you're wrong. ``Destroyed_CDVD2`` tracks only the cases where we explicitly destroyed the menu item. What if something else happens and cdvd2 dereferences null ? in that case the code won't enter the true branch.